src/main.rs

```diff
@@ -246,70 +310,98 @@ async fn main() -> Result<()> {
             addr,
             auth_token,
             key,
+            rpc_port,
         } => {
-            tokio::select! {
```

Comment:
why can't we select biased over this anymore, with just the ctrl_c having priority?

Reply:
The provider is now useful independent of the path you might or might not want to add. So you want to start it up as soon as possible, and then add the path or the stuff from stdin.

Basically instead of
- build the collection
- start the server

it is now
- start the server
- build the collection
- add it to the already running server

This is necessary because when you wait for stdin you might wait for a long time / forever, but we still want iroh to be immediately available for rpc calls.